src/solver/LeastSquareSolver.cpp

```diff
@@ -91,9 +58,15 @@ namespace abstract
     first_.clear();
     for (const auto& xi : variables())
     {
-      first_[xi.get()] = true;
+      first_[xi.get()] = {};
```

Comment:
Maybe we should rename `first_` into `bounds_`?

Reply:
The idea is to know which bounds is used first, when multiple bounds are put on the same variable. I renamed it as `boundsOrder_` in 2e48a737032e8255a966630a9b0940045093222c